home/tests/test_api.py

```diff
@@ -2,234 +2,244 @@
-        self.assertEqual(content["count"], 1)
+        assert content["count"] == 1
         # it should return only viber pages if filtered
-        response = self.client.get("/api/v2/pages/?viber=true")
+        response = uclient.get("/api/v2/pages/?viber=true")
         content = json.loads(response.content)
-        self.assertEqual(content["count"], 0)
+        assert content["count"] == 0
         # it should return all pages for no filter
-        response = self.client.get("/api/v2/pages/")
+        response = uclient.get("/api/v2/pages/")
         content = json.loads(response.content)
         # exclude home pages and index pages
-        self.assertEqual(content["count"], 3)
+        assert content["count"] == 3
 
-    def test_whatsapp_draft(self):
-        self.content_page2.unpublish()
-        page_id = self.content_page2.id
+    def test_whatsapp_draft(self, uclient):
+        """
+        Unpublished whatsapp pages are returned if the qa param is set.
+        """
+        page2 = ContentPage.objects.last()
+        page2.unpublish()
+        page_id = page2.id
         url = f"/api/v2/pages/{page_id}/?whatsapp=True&qa=True"
         # it should return specific page that is in draft
-        response = self.client.get(url)
+        response = uclient.get(url)
         content = json.loads(response.content)
         message = "*Self-help programs* 🌬️"
         # the page is not live but whatsapp content is returned
-        self.assertEqual(self.content_page2.live, False)
-        self.assertEqual(
-            content["body"]["text"]["value"]["message"].replace("\r", ""),
-            message,
-        )
+        assert not page2.live
+        assert content["body"]["text"]["value"]["message"].replace("\r", "") == message
 
-    def test_messenger_draft(self):
-        self.content_page2.unpublish()
-        page_id = self.content_page2.id
+    def test_messenger_draft(self, uclient):
+        """
+        Unpublished messenger pages are returned if the qa param is set.
+        """
+        page2 = ContentPage.objects.last()
+        page2.unpublish()
+        page_id = page2.id
         url = f"/api/v2/pages/{page_id}/?messenger=True&qa=True"
         # it should return specific page that is in draft
-        response = self.client.get(url)
+        response = uclient.get(url)
 
         message = "*Self-help programs* 🌬️"
         content = json.loads(response.content)
 
         # the page is not live but messenger content is returned
-        self.assertEqual(self.content_page2.live, False)
-        self.assertEqual(content["body"]["text"]["message"].replace("\r", ""), message)
+        assert not page2.live
+        assert content["body"]["text"]["message"].replace("\r", "") == message
+
+    def test_pagination(self, uclient):
+        """
+        FIXME:
+         * It's unclear what this is actually testing.
+         * Should it be multiple tests instead of just one?
+        """
```

Comment:
This looks to be testing getting multiple whatsapp messages for a single ContentPage, the `?message=...` query parameter.

I think it could definitely be split out into something like: whatsapp body when whatsapp disabled, first message when no `message` parameter specified, error if message index is greater than number of messages, error if message index is not an integer, and specific message if message query param is supplied.

Reply:
🤔

Reply:
I'm going to leave this as-is for now.